### PR TITLE
Fold fill into max

### PR DESCRIPTION
**Pull request by adam-smnk**
Adds pattern that folds linalg.fill into linalg.max and outputs combined linalg.generic.

A constant filled buffer is replaced by a single constant used directly in max operation on the elements of the other operand.
This allows to eliminate potential temporary buffer allocation and value initialization.

### CODE REVIEW

**Review by rengolin — COMMENTED** [review]
Why is this restricted to max and not just any element-wise op?

**Comment by adam-smnk** [comment]
> Why is this restricted to max and not just any element-wise op?

I think this could be easily relax to all named ops. Processing generics is also doable but hassle.
All in all, I just didn't want to bother for now.

**Comment by adam-smnk** [comment]
Fill folding together with broadcast folding finally improves named ops benchmarks.
The remaining slowdown is caused by one more temporary buffer allocation. It most likely caused by the folded max generic which is not in in-place format (outs to tensor.empty). The next step is to rewrite the max generic same as we do with `linalg-convert-add-in-place` for generic adds.

File: lib/TPP/Transforms/FoldIntoEltwise.cpp [hunk 1]
```diff
@@ -105,10 +105,93 @@ struct BroadcastIntoEltwise
...
+        inputs.push_back(input);
+        indexingMaps.push_back(
+            maxOp.getMatchingIndexingMap(maxOp.getDpsInputOperand(idx)));
+        continue;
+      }
+
+      // TODO: Check if there is any benefit in this over passing scalar
+      //       as an input argument.
+      // Store the constant separately to be later inserted directly into
+      // generic's body.
```
Comment (adam-smnk) [inline lib/TPP/Transforms/FoldIntoEltwise.cpp:155]: For now I wanted to have the constant inlined but actually I'm not sure if it makes any real difference.
If it doesn't, this pass could be simplified and easy generalized.

**Review by rengolin — APPROVED** [review]

**Comment by adam-smnk** [comment]
Looks like the two folders could be merged, it just requires more testing to see if I missed any edge cases.
I'll follow up on that a bit later.
